dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/GoogleAI/GoogleAIEmbeddingClient.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI.Core;
+
+/// <summary>
+/// Represents a client for interacting with the embeddings models by Google AI.
+/// </summary>
+internal sealed class GoogleAIEmbeddingClient : ClientBase
+{
+    private readonly string _embeddingModelId;
+    private readonly Uri _embeddingEndpoint;
+
+    /// <summary>
+    /// Represents a client for interacting with the embeddings models by Google AI.
+    /// </summary>
+    /// <param name="httpClient">HttpClient instance used to send HTTP requests</param>
+    /// <param name="modelId">Embeddings generation model id</param>
+    /// <param name="apiKey">Api key for GoogleAI endpoint</param>
+    /// <param name="logger">Logger instance used for logging (optional)</param>
+    public GoogleAIEmbeddingClient(
+        HttpClient httpClient,
+        string modelId,
+        string apiKey,
+        ILogger? logger = null)
+        : base(
+            httpClient: httpClient,
+            logger: logger)
+    {
+        Verify.NotNullOrWhiteSpace(modelId);
+        Verify.NotNullOrWhiteSpace(apiKey);
+
+        this._embeddingModelId = modelId;
+        this._embeddingEndpoint = new Uri($"https://generativelanguage.googleapis.com/v1beta/models/{this._embeddingModelId}:batchEmbedContents?key={apiKey}");
```

Comment:
The `v1beta` part of the URL is going to change and there may be times when multiple versions will be supported at the same time.

We should have:

1. A central place where the default API version is set
2. We should consider if we want to allow developers to change this. We do allow this in Azure OpenAI. It's a risky thing for a developer to do but that doesn't mean someone may have a good reason and we don't necessarily want to block them.

Reply:
So create static class or inject api version via DI extensions?

Reply:
Created #5659 for tracking of this.